### PR TITLE
Pipeline Updates

### PR DESCRIPTION
**Pull request by wilkermichael**
If you go through my PR commit by commit it's easier to review, I typically add extra context and you won't see things change between commits unintentionally 

Changes proposed in this PR:
- Upgrade Go version to 1.20 and specifically 1.20.1 to address a CVE
- Updated `gotestsum` calls to use `testnames` this improves debugging as you'll see which tests pass and fail, not just the package name with failing/passing tests
- Added `1.1.x` release branch for nightly testing

How I've tested this PR:
I verified that the nightly tests passed:
image

How I expect reviewers to test this PR:
👀 

Checklist:
- [N/A] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: control-plane/subcommand/common/common_test.go [hunk 1]
```diff
@@ -218,7 +216,6 @@ func TestWriteFileWithPerms_InvalidOutputFile(t *testing.T) {
 
 func TestWriteFileWithPerms_OutputFileExists(t *testing.T) {
 	t.Parallel()
-	rand.Seed(time.Now().UnixNano())
```
Comment (wilkermichael) [inline control-plane/subcommand/common/common_test.go:221]: rand.Seed has been deprecated. rand now automatically gets seeded at startup

File: cli/cmd/troubleshoot/upstreams/upstreams.go [hunk 2]
```diff
@@ -245,6 +246,9 @@ func formatClusterNames(names map[string]struct{}) string {
 	for k := range names {
 		out = append(out, k)
 	}
+
+	// Sort cluster names first before outputting to guarantee order
+	sort.Strings(out)
```
Comment (wilkermichael) [inline cli/cmd/troubleshoot/upstreams/upstreams.go:251]: Iryna fixed this as well, so i'll drop this commit and rebase once she merges in her change

**Review by thisisnotashwin — APPROVED** [review]
This looks great. Have a comment that is not related to the 1.20 change. Great work on this.

File: control-plane/subcommand/common/common_test.go [hunk 3]
```diff
@@ -163,36 +162,36 @@ func TestConsulLogin_TokenNotReplicated(t *testing.T) {
 
 func TestConsulLogin_EmptyBearerTokenFile(t *testing.T) {
 	t.Parallel()
-	require := require.New(t)
+	req := require.New(t)
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply control-plane/subcommand/common/common_test.go:165]: yeah, that's better! I'll do that

**Review by thisisnotashwin — APPROVED** [review]

**Pull request merged**